### PR TITLE
Only display clear button in notifications modal when there are notifcations

### PR DESCRIPTION
**Pull request by zachariast**
Fix for #303

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -299,7 +299,7 @@ ul.no_bullet_list li {
 }
 
 .display_none {
-  display: none;
+  display: none !important;
```
Comment (Wowu) [inline app/assets/stylesheets/application.scss]: There is a problem with this line, apparently Capybara (library for testing code) doesn't recognise elements that have `display: none !important`. Are you sure we need `!important` property here?

File: app/views/shared/_modal.html.erb [hunk 2]
```diff
@@ -2,16 +2,21 @@
...
-		<% elsif local_assigns[:modal_close_class] %>
-			<h1 class=<%= "#{local_assigns[:modal_close_class].to_s} align_right" %>><i class="fa fa-times"></i></h1>
-		<% elsif local_assigns[:modal_close_id] %>
-			<h1 id=<%= local_assigns[:modal_close_id].to_s %> class="align_right"><i class="fa fa-times"></i></h1>
-		<% end %>
-	</div>
+		<div class="align_right">
+			<% if local_assigns[:modal_clear] %>
+			<% if Notification.all.size >= 1 %>
+				<h1 id="clear_notifcations" class="align_left small_margin_right"><%= local_assigns[:modal_clear] %></h1>
```
Comment (Wowu) [inline app/views/shared/_modal.html.erb]: Indents here are a bit unclear.

File: app/assets/javascripts/notifications.js [hunk 3]
```diff
@@ -238,8 +238,10 @@ var onReadyNotifications = function() {
     emptyNotificationsList();
     showNotificationsNone();
     changeTitle(0);
+    $('#notifications #notifications_text .align_right h1#clear_notifcations').hide();
```
Comment (julianguyen) [inline app/assets/javascripts/notifications.js]: You could possibly get away with just using `$(#notifications #clear_notifications).hide()`

File: app/views/shared/_modal.html.erb (hunk 2 above)
Comment (julianguyen) [inline app/views/shared/_modal.html.erb]: Let's stick with 2 space idents :)

**Comment by julianguyen** [comment]
Great work @zachariast, update this PR with the comments suggested, and it's good to go. Currently and unfortunately :'( there are no test specs for Notifications.js. If you're interested, you could work on this next!

**Comment by zachariast** [comment]
@julianguyen yess, i'll work on notifications tests. i'll open an issue during the day.

**Comment by julianguyen** [comment]
LGTM! Please squash your commits and merge when you're ready!

**Pull request merged**